feat(client): ✨ Add CreateLobby to Dev

mrjoshua520 opened the pull request.

zaverbac committed:
feat(client): ✨ Add basic UI for create lobby

Add basic UI for create lobby page with button, player list, player amount, and lobby code Co-authored-by: Joshua Salkeld mrjoshua520@users.noreply.github.com

mrjoshua520 committed:
Merge branch 'development' into createlobby

zaverbac committed:
feat(client): ✨ Create lobby for users to join

Create lobby that adds lobby code, and allows users to join
Co-authored-by: Josh Salkeld <mrjoshua520@users.noreply.github.com>

zaverbac committed:
Merge branch 'development' of https://github.com/cat-acronym-team/cat-me-if-you-can into createlobby

zaverbac committed:
refactor(client): ✨ Retrieve user uids, generate lobby and lobby code

Retrieve user id from current user and create lobby and ability to genreate new lobby code with lobby
Co-authored-by: Joshua Salkeld <5087921+mrjoshua520@users.noreply.github.com>

zaverbac committed:
refactor(client): ✨ Retrieve user uids, generate lobby and lobby code

Retrieve user id from current user and create lobby and ability to genreate new lobby code with lobby
Co-authored-by: Joshua Salkeld <5087921+mrjoshua520@users.noreply.github.com>

zaverbac committed:
Merge branch 'createlobby' of https://github.com/cat-acronym-team/cat-me-if-you-can into createlobby

mrjoshua520 committed:
feat(client): ✨ Generate Code and Invite URl

Generates a five character code and displays it on the UI. Also displays the invite link below the start button on the UI

Co-authored-by: Zachary Averbach <zaverbac@users.noreply.github.com>
Co-authored-by: Giantte Chambers <gchamb@users.noreply.github.com>

mrjoshua520 committed:
refactor(client): 🎨 Change lobby page to a component

The svelte file for displaying the lobby has been moved to a component.

Co-authored-by: Zachary Averbach <zaverbac@users.noreply.github.com>

mrjoshua520 committed:
refactor(client): 🎨 Change Comments

Changes some comments to follow the TODo format

nstringham committed:
refactor(client): 🚚 rename createlobby to create-lobby

because kabab case is standard for web based projects

nstringham committed:
refactor(client): 🔥 remove unused code

nstringham commented on client/src/lib/firebase/create-lobby.ts:
```diff
@@ -0,0 +1,49 @@
+    code += characters.charAt(Math.floor(Math.random() * charactersLength));
+  }
+
+  const checkCode = await getDocs(lobbyCollection);
```
let's not download the entire lobbies collection :)

mrjoshua520 committed:
perf(client): ⚡️ Improve performance when creating lobby code

Improves the performance by not pulling every document that exists

mrjoshua520 committed:
Merge branch 'development' of https://github.com/cat-acronym-team/cat-me-if-you-can into createlobby

mrjoshua520 committed:
refactor(client): 🔥 Remove unused image files

Removes unused google image files from the sign in folder

mrjoshua520 committed:
refactor(client): 🎨 Change user signed in If statement

zaverbac committed:
feat(client): ✨ User logs in anonymous in lobby

User logs in anonymously when create lobby button is pressed if use is currently not logged in

mrjoshua520 committed:
Merge branch 'development' into createlobby

mrjoshua520 committed:
feat(client): ✨ Connect splash page with create lobby

The splash page is now capable of creating a lobby with the create lobby button

mrjoshua520 committed:
refactor(client): 🎨 Add export for user collection

Added an export for the user collection in the firebase collections file and updated code accordingly

nstringham commented on client/src/routes/+page.svelte:
```diff
@@ -39,11 +40,10 @@
 
-    // TODO: Push to game page with code
-    goto("/");
+    const code = (await createLobby()) as string;
```
lets not have the `as` here because is isn't needed

nstringham replied:
@mrjoshua520
